.github/workflows/compliance-ncs.yml

```diff
@@ -0,0 +1,63 @@
+name: Compliance NCS
+
+on: pull_request
+
+jobs:
+  compliance_job:
+    runs-on: ubuntu-latest
+    name: Run compliance checks on patch series (PR)
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v2
+        with:
+          path: ncs/zephyr
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: cache-pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-doc-pip
+
+      - name: Run Compliance Tests
+        env:
+          BASE_REF: ${{ github.base_ref }}
+        working-directory: ncs
+        run: |
+          pip3 install -U pip
+          pip3 install -U setuptools
+          export PATH="$HOME/.local/bin:$PATH"
+          pip3 install -U west
+          west init -m https://github.com/nrfconnect/sdk-nrf
+          west update zephyr nrfxlib
+          pip3 install -U gitlint
+          pip3 install -r zephyr/scripts/requirements-compliance.txt
+          export ZEPHYR_BASE=$(pwd)/zephyr
```

Comment:
Why not set `working-directory: ncs/zephyr` and then:

```
source zephyr-env.sh
```

Reply:
I don't really know if running `west init -m https://github.com/nrfconnect/sdk-nrf` inside `zephyr` works OK but if it does I can update it. The only requirement for running `Kconfig` compliance is to have `ZEPHYR_BASE` pointing to the correct place, so I don't know what purpose the extra `zephyr-env.sh` lines would add here (and the other compliance checks are all local). Also I messed up my conflict resolution because there was a `west update -k` in before I pushed, will fix now.

Reply:
`source zephyr-env.sh` <- Done